### ISSUE TITLE
[WebToolsE2E][Aspire] Building .NET MAUI App with ".NET Aspire Orchestrator Support" fails with error MSB4057: The target "GetTargetPath" does not exist in the project.

### ISSUE THREAD

**Issue by v-elenafeng** [dotnet/aspire#3043]
**REGRESSION INFO** 

Reproduces with VS 17.10 Preview 3 (Includes Aspire 8.0 P4) + Aspire 8.0 P5

**INSTALL STEPS**

Clean machine: Win11 x64 23h2 ENU

1. Install VS 17.10 Preview 3 from VS/main

   - Web Workload + Maui workload
   - Includes Aspire 8.0 Preview 4
   - Includes SDK 8.0.300-preview.24122.6

1. Create .NET MAUI App 8.0 and build

1. Right click the project > Add ".NET Aspire Orchestrator Support" > Build

**ACTUAL**
error MSB4057: The target "GetTargetPath" does not exist in the project.
![image](https://github.com/dotnet/aspire/assets/78580199/fabef1a6-580d-4701-9405-4b68e9a5f1a8)

This issue also reproduces with "Only Maui workload"

**EXPECTED**
The .NET MAUI App can be built successfully with ".NET Aspire Orchestrator Support".

**Comment by eerhardt** [comment]
I don't have MAUI installed locally. Can someone attach a [binlog](https://msbuildlog.com/) of the failing build, so we can see where it is coming from? That link has instructions on how to enable it for VS.

**Comment by v-elenafeng** [comment]
> I don't have MAUI installed locally. Can someone attach a [binlog](https://msbuildlog.com/) of the failing build, so we can see where it is coming from? That link has instructions on how to enable it for VS.

Unable to attach the file type, it can be found here "\\vwdbuild01\temp\Elena Feng\v-elenafeng\Issue#3043\MauiApp4"

**Comment by eerhardt** [comment]
@v-elenafeng - the binlog at that location is for a build that succeeded. Can you grab the log for a build that fails with the above error?

**Comment by v-elenafeng** [comment]
> @v-elenafeng - the binlog at that location is for a build that succeeded. Can you grab the log for a build that fails with the above error?

I grabbed the binlog after the build in VS showed a failure, but it wasn't captured in the logs, so I ran "dotnet build" the solution using CLI and the build was successful as well. So I will move this to DevOps since it's like a VS tooling issue.

Btw, I deleted the comment posted above since it's an internal shared path.

**Comment by BillHiebert** [comment]
Need to investigate this.

**Comment by BretJohnson** [comment]
Note that this issue looks to affect all multitargetted projects, not just MAUI. I created a Console app with `<TargetFrameworks>net9.0;net8.0</TargetFrameworks>` and saw the same "The target "GetTargetPath" does not exist in the project" error.

**Comment by v-elenafeng** [comment]
Yes, when adding '.NET Aspire Orchestrator Support' to a multi-target project, it's keeps the AppHost single-targeted, so building the project throws an error. This can be fixed by modifying the AppHost project to be multi-targeted.

![image](https://github.com/user-attachments/assets/4da1a73a-56c5-4de2-ba33-d2ef11f589e2)
![image](https://github.com/user-attachments/assets/0ffcb39a-d319-4c46-bd30-926d2fadb807)

**Comment by BillHiebert** [comment]
I don't think multi-targeting AppHost is the right thing to do here. Since AppHost does not reference types from the referenced projects,  I think it is better is to add the PrivateAssets="All" and ReferenceOutputAssembly="false" attributes to the project reference.

**Comment by DrLeh** [comment]
I encountered this today. if the aspire host uses `<TargetFrameworks>` (plural), even with only one target I get this error. I changed it to `<TargetFramework>` (singular) and it went away

**Comment by papuashu** [comment]
I ran into same problem as @v-elenafeng ... I got around it by targeting Windows runtime only in my Blazor Hybrid (MAUI) project and using TargetFramework (singular) element. It's fine as long as I target Windows only; probably a separate build / deploy for MAUI multi-targeted project would do it.

**Comment by ntomasn** [comment]
The problem with multi-targeting projects and the App Host can be solved using a MSBuild Target Element.

```
<Project Sdk="Microsoft.NET.Sdk.Web">
	<PropertyGroup>
		<TargetFrameworks>net10.0</TargetFrameworks>
	</PropertyGroup>

	<Target Name="GetTargetPath" Returns="net10.0" />
</Project>
```

**Comment by Schinwinkwinsky** [comment]
I faced this problem today. Remove the Maui.Client project reference from AppHost project. 
The Aspire docs alert about that: [Aspire Maui Integration](https://aspire.dev/integrations/dotnet/maui/#add-net-maui-project)
The Maui docs need to be updated: [.Net Maui integration to Aspire](https://learn.microsoft.com/pt-br/dotnet/maui/data-cloud/aspire-integration?view=net-maui-10.0)

**Comment by dosolkowski-work** [comment]
@BillHiebert's original suggestion of changing the project references did not work. Adding `<Target Name="GetTargetPath" Returns="net10.0" />` to the project files of the referenced projects did stop the error, but somehow even with that in place and using `.WithArgs("--framework", "net10.0")` when adding a project to a host, Aspire is still running the .NET 8 version instead of the .NET 10 version. In combination with #17912, it seems like multi-targeting of any kind in combination with Aspire is not supported.
